pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -0,0 +1,859 @@
+        """Control the instrument step generator invert option (boolean).""",
+        validator=strict_discrete_set,
+        values={True: 'ON', False: 'OFF'},
+        map_values=True,
+        get_process=lambda r:
+        r[3].split(":")[1] == 'ON' if isinstance(r, list)  # case stepgen output 'ON'
+        else r.split(":")[1]
+    )
+
+    stepgen_step_size_multiplier = Instrument.control(
+        "STPgen?", "STPgen OFFset:%f",
+        """Control the step multiplier value which is the offset (float).""",
+        get_process=lambda r:
+        float(r[2].split(":")[1]) if isinstance(r, list)  # case stepgen output 'ON'
+        else r.split(":")[1]
+    )
+
+    stepgen_step_source_and_size = Instrument.control(
+        "STPgen?", "STPgen %s:%s",
+        """Control the step size and souce for the step generator (Amps or Volts per step).\n
+        Use example: instrument.stepgen_step_source_and_size = ("VOLTAGE", 2.0)
+        Use example: instrument.stepgen_step_source_and_size will return ['VOLTAGE', 2.0]
+        """,
+        # e.g. if we want to configure the step generator as voltage with step size equals to 2V
+        # then we will use the next expression: instrument.stepgen_step_source_and_size = (
+        # "VOLTAGE", 2.0).\n After that, if we want to ask the instrument for the step generator
+        # source and size then we will use the next expression:
+        # instrument.stepgen_step_source_and_size and we should obtain a list as follow:
+        # ['VOLTAGE',2.0]. Parameters out ouf range will have no effect.
+        get_process=lambda r:
+        [r[5].split(":")[0], float(r[5].split(":")[1])] if isinstance(r, list)
+        else r.split(":")[1]
+    )
+
+    def get_stepgen_offset(self):
+        """
+
+        :return:
+        """
+        mult = self.stepgen_step_size_multiplier
+        s_size = self.stepgen_step_source_and_size[1]
+        invert = self.stepgen_invert
+        abs_offset = mult * s_size
+        return -abs_offset if invert else abs_offset
+
+    def set_stepgen_offset(self, offset):
+        """
+        Configure the offset of the step generator of the instrument
+        changing the step size multiplier and invert param.
+        :param offset: Will be a float valid between some valid range which depends on the step size
+        and step generator source. The instrument only will consider increments of the 10%
+        of the step size, other values will not have effect on the step generator offset.
+        """
+        stepgen_step_source_and_size = self.stepgen_step_source_and_size
+        stepgen_step_size = stepgen_step_source_and_size[1]
+        stepgen_step_source = stepgen_step_source_and_size[0]
+        max_offset = 5.0 * stepgen_step_size
+
+        if abs(offset) > max_offset:
+            offset_units = "V" if "VOLTAGE" in stepgen_step_source else "A"
+            raise ValueError(
+                f'Setting the step generator offset....\n'
```

Comment:
You can turn it into a plain python property:
~~~python
@property
def stepgen_offset(self):
    """Control the step generator offset."""  # docstring for the property
    return some_value  # getter part

@stepgen_offset.setter
def stepgen_offset(self, value):
    pass  # setter part
~~~

Reply:
Done. Thanks